backend/fms_core/services/container.py

```diff
@@ -13,41 +15,87 @@ def get_container(barcode=None):
-            **container_data,
-            defaults={'comment': f"{comment}",
-                      'name': name or barcode},
+    if barcode:
+        container_data = dict(
+            **(dict(location=container_parent) if container_parent else dict(location__isnull=True)),
+            **(dict(barcode=barcode) if barcode is not None else dict()),
+            **(dict(name=name) if name is not None else dict(name=barcode)), # By default, a container name will be his barcode
+            **(dict(coordinates=coordinates) if coordinates is not None else dict()),
+            **(dict(kind=kind) if kind is not None else dict()),
         )
 
-        if not is_container_created and container.kind == "tube":
-            warnings.append(f"Using existing Container with barcode '{container.barcode}'.")
+        #TODO: check sample or container creation templates where only barcode OR name is required
+
+        comment = creation_comment or (f"Automatically generated on {datetime.utcnow().isoformat()}Z")
+
+        try:
+            container = Container.objects.get(barcode=barcode)
+
+            # Validate that the retrieved container is the right one
+            if kind and kind != container.kind:
+                errors.append(f"Provided container kind {kind} does not match the container kind {container.kind} of the container retrieved using the barcode {barcode}.")
+            if name and name != container.name:
+                errors.append(f"Provided container name {name} does not match the container name {container.name} of the container retrieved using the barcode {barcode}.")
+            if container_parent and container_parent.id != container.location.id:
+                errors.append(f"Provided parent container {container_parent.barcode} does not match the parent container {container.location.barcode} of the container retrieved using the barcode {barcode}.")
+            if coordinates and coordinates != container.coordinates:
+                errors.append(f"Provided container coordinates {coordinates} do not match the container coordinates {container.coordinates} of the container retrieved using the barcode {barcode}.")
+
+        except Container.DoesNotExist:
+            if container_parent and CONTAINER_KIND_SPECS[container_parent.kind].requires_coordinates and not coordinates:
+                errors.append(f"Parent container kind {container_parent.kind} requires that you provide coordinates.")
+            else:
+                try:
+                    container = Container.objects.create(**container_data, comment=comment)
+                    created_entity = True
+                # Pile up all validation error raised during the creation of the container
+                except ValidationError as e:
+                    errors.append(';'.join(e.messages))
+    else:
+        errors.append(f"Barcode is required to get or create a container.")
+
+    return (container, created_entity, errors, warnings)
```

Comment:
Good idea to have such an indicator. What would you think about generalizing this kind of indicator so that all the service functions return it? Could be useful

Reply:
I mean the created_entity indicator. So all the services would return if the item was created, updated, or retrieved for example with a string

Reply:
This is specifically for the get_or_create so the business end can know if this is a new entity or a fetched one. We should indeed add this indicator to all get_or create.

As for it being a string: no. This is only to differentiate between those two options. It already complicates the error handling anymore than this and it will start to be more an hindrance than anything.

Reply:
Ok that works for me!